src/libs/actions/Report.js

```diff
@@ -323,6 +333,81 @@ function fetchChatReportsByIDs(chatList) {
         });
 }
 
+/**
+ * Given IOU and chat report ID fetches most recent IOU data from API and update stored data in Onyx.
+ *
+ * @param {Number} iouReportID
+ * @param {Number} chatReportID
+ */
+function updateIOUReportData(iouReportID, chatReportID) {
+    API.Get({
+        returnValueList: 'reportStuff',
+        reportIDList: iouReportID,
+        shouldLoadOptionalKeys: true,
+        includePinnedReports: true,
+    }).then((response) => {
+        if (!response) {
+            return;
+        }
+
+        if (response.jsonCode !== 200) {
+            console.error(response.message);
+            return;
+        }
+
+        const iouReportData = response.reports[iouReportID];
+        if (!iouReportData) {
+            console.error(`No iouReportData found for reportID ${iouReportID}`);
+            return;
+        }
+        const iouReportObject = getSimplifiedIOUReport(iouReportData, chatReportID);
```

Comment:
We are duplicating again... 

https://github.com/Expensify/Expensify.cash/blob/f64e2f3b95ba15666f6963ae42cce2fe537e2719/src/libs/actions/Report.js#L251-L272

Mostly the same code.

Reply:
I just submitted the commit for these changes.